File: Codeforces/combinatorics/binomial_coeff/2025B.py
 

ks = [0] * 100000
power_of_twos = [1] * 100000

moduo = 1000000007

def mul(x, y):
    return (x*y)%moduo if x * y >= moduo else x * y

def add(x, y):
    return (x+y)%moduo if x + y >= moduo else x + y


def main():
    t = int(input())

    ns = list(map(int, input().split()))
    ks = list(map(int, input().split()))


    for i in range(1,100000):
        power_of_twos[i] = mul(2, power_of_twos[i-1])


    for i in range(t):
        if ns[i] == ks[i]:
            print(1)
        else:
            print(power_of_twos[ks[i]])



if __name__ == "__main__":
    main()

File: Codeforces/combinatorics/binomial_coeff/2025B.cpp
#include <vector>
#include <numeric>
#include <algorithm>
#include <iostream>

using namespace std;

int ns[100000];
int ks[100000];

unsigned long long moduo = 1000000007;

unsigned long long mul(unsigned long long x, unsigned long long y)
{
    return x * y >= moduo ? (x * y) % moduo : x * y;
}
unsigned long long add(unsigned long long x, unsigned long long y)
{
    return x + y >= moduo ? (x + y) % moduo : x + y;
}
unsigned long long power_of_twos[100000];

int main()
{
    int t;

    cin >> t;

    power_of_twos[0] = 1;
    for (int i = 1; i < 100000; i++)
        power_of_twos[i] = mul(2, power_of_twos[i - 1]);


    for (int i = 0; i < t; i++)
        cin >> ns[i];

    for (int i = 0; i < t; i++)
        cin >> ks[i];


    for (int i = 0; i < t; i++)
    {
        if (ns[i] == ks[i])
            cout << 1 << endl;
        else
            cout << power_of_twos[ks[i]] << endl;
    }

    


    return 0;
}
